packages/react/src/components/List/HierarchyList/HierarchyList.test.jsx

```diff
@@ -830,32 +830,101 @@ describe('HierarchyList', () => {
+    const onSelect = jest.fn();
+    const { container } = render(
+      <HierarchyList
+        items={items}
+        title="Hierarchy List"
+        editingStyle={EditingStyle.MultipleNesting}
+        onSelect={onSelect}
+      />
+    );
 
-    afterEach(() => {
-      jest.resetAllMocks();
-    });
+    userEvent.click(screen.getByTestId('Atlanta Braves-checkbox'));
+    expect(onSelect).toHaveBeenCalled();
+    expect(screen.getByText('10 items selected')).toBeVisible();
+    userEvent.click(screen.getByTestId('Atlanta Braves-checkbox'));
+    expect(onSelect).toHaveBeenCalled();
+    expect(screen.queryByText('10 items selected')).toBeNull();
+    expect(container.querySelectorAll('input[checked]').length).toBe(0);
+  });
 
-    it('should maintain the current page when page size changes', () => {
-      const listItems = getListItems(20);
-      const { rerender } = render(
-        <HierarchyList title="Test List" items={listItems} pageSize="lg" isVirtualList />
-      );
+  it('should not selected a locked row when parent selected', () => {
+    const onSelect = jest.fn();
+    render(
+      <HierarchyList
+        items={items}
+        title="Hierarchy List"
+        editingStyle={EditingStyle.MultipleNesting}
+        lockedIds={['Atlanta Braves_Dansby Swanson']}
+        defaultExpandedIds={['Atlanta Braves']}
+        onSelect={onSelect}
+      />
+    );
 
-      expect(screen.getByText('Item 9')).toBeVisible();
-      userEvent.click(screen.getByRole('button', { name: 'Next page' }));
-      expect(screen.getByText('Item 19')).toBeVisible();
+    userEvent.click(screen.getByTestId('Atlanta Braves-checkbox'));
+    expect(onSelect).toHaveBeenCalled();
+    expect(screen.getByText('9 items selected')).toBeVisible();
+    userEvent.click(screen.getByTestId('Atlanta Braves-checkbox'));
+    expect(onSelect).toHaveBeenCalled();
+    expect(screen.queryByText('9 items selected')).toBeNull();
+    expect(screen.getByTestId('Atlanta Braves_Dansby Swanson-checkbox')).not.toBeChecked();
+  });
 
-      rerender(<HierarchyList title="Test List" items={listItems} pageSize="sm" isVirtualList />);
+  it('should set indeterminate state on parent when a child is checked', () => {
+    render(
+      <HierarchyList
+        items={items}
+        editingStyle={EditingStyle.MultipleNesting}
+        defaultExpandedIds={['Atlanta Braves']}
+        title="indeterminate Ids"
+      />
+    );
 
-      expect(screen.getByText('Page 2')).toBeVisible();
-      expect(screen.getByText('Item 9')).toBeVisible();
-    });
+    userEvent.click(screen.getByTestId('Atlanta Braves_Ronald Acuna Jr.-checkbox'));
+    expect(screen.getByTestId('Atlanta Braves-checkbox')).toBePartiallyChecked();
+    userEvent.click(screen.getByTestId('Atlanta Braves_Ronald Acuna Jr.-checkbox'));
+    expect(screen.getByTestId('Atlanta Braves-checkbox')).not.toBeChecked();
+  });
+
+  it('should unset indeterminate state on parent when a child is unchecked', () => {
```

Comment:
"should unset indeterminate state on parent when a PRESELECTED child is unchecked"
just testing the uncheck of a child is already tested in the test above

Reply:
renamed